_posts/2017-03-04-bettracker-typescript.md

```diff
@@ -0,0 +1,108 @@
+---
+layout:     post
+title:      Modernizing our app with TypeScript
+date:       2017-03-18 09:00
+summary:    How we used TypeScript to successfully refactor a large business critical codebase.
+category:   Software Engineering
+tags:       TypeScript, react, re-factoring
+author:     james_moorhouse
+---
+
+# Introduction
+
+Back in October 2016 my team were looking for ways we could improve our My Bets product which was written using ReactJS. We'd first made the product about 2 and a half years ago, much before my time, when ReactJS was still fairly new on the scene. As such the ReactJS itself has been through quite a few changes since then and a few of our design patterns were in need of a spring clean.
+
+As a product, My Bets allows our customers to view, track and cashout any of their bets. Bets these days are complex, they can consist of many selections consisting of multiple outcomes across multiple events - oh, and then we need to track the history of a bet i.e. all the data at the time it was placed and compare it to the data of the bet, if you were to place the same one now.
+
+![My Bets Overview](/images/mybets-TypeScript/mybets-overview.png)
```

Comment:
the case is incorrect in all of these image links.  It should be `images/mybets-typescript/`